pkg/config/config.go

```diff
@@ -83,14 +88,73 @@ func (c *Config) ConfigureRules() {
 }
 
 func loadConfig(filename string) (c Config, err error) {
+	if isValidURL(filename) {
+		// if fileName is a valid URL, we will download and set it to the config
+		log.Debug().Str("url", filename).Msg("Downloading file from")
+		// hardcoding this file and saving to root directory
+		downloadedFile := "downloadedRules.yaml"
+		err := DownloadFile(downloadedFile, filename)
+		if err != nil {
+			return c, err
+		}
+		filename = downloadedFile
+		log.Debug().Str("filename", filename).Msg("Saved remote config to local file.")
+	}
+
 	yamlFile, err := ioutil.ReadFile(filename)
+	log.Debug().Str("filename", filename).Msg("Adding custom ruleset from")
 	if err != nil {
 		return c, err
 	}
-
 	return c, yaml.Unmarshal(yamlFile, &c)
 }
 
+// isValidUrl tests a string to determine if it is a valid URL or not
+func isValidURL(toTest string) bool {
+	_, err := url.ParseRequestURI(toTest)
+	if err != nil {
+		return false
+	}
+
+	u, err := url.Parse(toTest)
```

Comment:
ParseRequestURI() is our primary basic check, but it will allow strings like ‘Test: Test’ to pass (which we don’t want). We therefore combine it with a url parser to check that both a scheme (like http) and a host exist.

Reply:
Got it, thanks for clarifying!